### ISSUE TITLE
Inline tests are not being run properly

### ISSUE THREAD

**Issue by copy** [ocaml/dune#1864]
Consider this test setup:

```
(library
  (name my_library)
  (inline_tests)
  (preprocess (pps ppx_jane))
  (libraries base))
```

```ocaml
let%test _ = false
```

<hr>

```
dune runtest --force
```

doesn't print anything, while manually invoking the inline test runner works:

```
% _build/default/.my_library.inline-tests/run.exe inline-test-runner my_library
File "foo.ml", line 1, characters 0-18: <<false>> is false.

FAILED 1 / 1 tests
```

stracing `dune runtest` reveals that the library name is not correctly expanded (dune invokes the test runner with the string `"${library-name}"`):

```
execve(".my_library.inline-tests/run.exe", [".my_library.inline-tests/run.exe", "inline-test-runner", "${library-name}", "-source-tree-root", ".", "-diff-cmd", "-"] …
```

Tested in dune 1.7.2.

**Comment by copy** [comment]
Now that I look at it again, it's probably due to the prefix change from `$` to `%` for dune variables which [has already been adjusted](https://github.com/janestreet/ppx_inline_test/blob/master/src/dune) in ppx_inline_test. Slightly unfortunate that neither dune nor ppx_inline_test complain about.

**Comment by diml** [comment]
Dune silently upgrades old `$` varialbes to `%` ones when dealing with jbuilder projects. However, it cannot upgrade unknown `$` variables since that was the old semantic: unknown `$` variables where left unexpanded. This was obviously bad and we changed that for `%` variables: unknown `%` variables cause an error and literal `%` characters must be escaped.

`library-name` wasn't considered for upgrade in dune <1.7.2. We fixed this in dune 1.7.2. however, because this upgrade happens at compilation time in ppx_inline_test, you need to reinstall ppx_inline_test after upgrading to dune 1.7.2.

Sorry about this, I know it's not ideal.
